src/api/quotes.rs

```diff
@@ -0,0 +1,259 @@
+// Copyright (C) 2020-2022 Daniel Mueller <deso@posteo.net>
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+use chrono::Date;
+use chrono::DateTime;
+use chrono::serde::ts_milliseconds::deserialize as datetime_from_timestamp;
+use chrono::Utc;
+use num_decimal::Num;
+use serde::Deserialize;
+
+use crate::api::aggregates::TimeSpan;
+use crate::api::response::Response;
+use crate::Str;
+
+/// Filters quote data based on the timestamp in the given direction of time.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QuoteTimespanFilter {
+    /// Filter for data older than the given timestamp.
+    LessThan,
+    /// Filter for data older than, or as old as, the given timestamp.
+    LessThanEqual,
+    /// Filter for data newer than the given timestamp.
+    GreaterThan,
+    /// Filter for data newer than, or as new as, the given timestamp.
+    GreaterThanEqual,
+}
+
+impl AsRef<str> for QuoteTimespanFilter {
+    fn as_ref(&self) -> &'static str {
+        match *self {
+            QuoteTimespanFilter::LessThan => "lt",
+            QuoteTimespanFilter::LessThanEqual => "lte",
+            QuoteTimespanFilter::GreaterThan => "gt",
+            QuoteTimespanFilter::GreaterThanEqual => "gte",
+        }
+    }
+}
+
+/// Specifies the sort order of the quotes, using the key specified in `sort`.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QuoteOrder {
+    /// Order quotes oldest to newest.
+    Ascending,
+    /// Order quotes newest to oldests.
+    Descending,
+}
+
+impl AsRef<str> for QuoteOrder {
+    fn as_ref(&self) -> &'static str {
+        match *self {
+            QuoteOrder::Ascending => "asc",
+            QuoteOrder::Descending => "desc",
+        }
+    }
+}
+
+/// Specifies the sort key of the quotes, especially relevant for ordering (see `QuoteOrder`).
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum QuoteSortBy {
+    /// Sort quotes by timestamp.
+    Timestamp,
+}
+
+impl AsRef<str> for QuoteSortBy {
+    fn as_ref(&self) -> &'static str {
+        match *self {
+            QuoteSortBy::Timestamp => "timestamp",
+        }
+    }
+}
+
+/// A GET request to be made to the
+/// `/v3/quotes/<symbol>` endpoint.
+#[derive(Clone, Debug, PartialEq)]
+pub struct QuotesReq {
+    /// The ticker symbol to request quote data for.
+    pub symbol: String,
+    /// A date with the format YYYY-MM-DD or a nanosecond timestamp.
+    pub timestamp: String,
```

Comment:
I don't think we should work with strings here. This should be a `Date` (or something more elaborate; but I am not sure we will need it nanosecond logic here).